client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -543,6 +543,26 @@ export class PluginsBrowser extends Component {
 	}
 }
 
+/**
+ * Filter the popular plugins list.
+ * Remove the displayed featured plugins from the popular list to
+ * avoid show them twice
+ *
+ * @param {Array} popularPlugins
+ * @param {Array} featuredPlugins
+ */
+function filterPopularList( popularPlugins = [], featuredPlugins = [] ) {
+	const displayedFeaturedSlugs = featuredPlugins
+		.slice( 0, SHORT_LIST_LENGTH ) // only displayed plugins
+		.map( ( plugin ) => plugin.slug );
+
+	const displayedFeaturedSlugsMap = new Map(
+		displayedFeaturedSlugs.map( ( slug ) => [ slug, slug ] )
+	);
+
+	return popularPlugins.filter( ( plugin ) => ! displayedFeaturedSlugsMap.has( plugin.slug ) );
```

Comment:
Instead of creating two new map can would the following work?
```suggestion
  return popularPlugins.filter(
		( plugin ) =>
			! featuredPlugins
				.slice( 0, SHORT_LIST_LENGTH )
				.some( ( featuredPlugin ) => featuredPlugin.slug === plugin.slug )
	);
```

Reply:
Yes, it is indeed an option. 

My main idea with creating a map was to build an O(n) instead of an O(n^2) solution. Currently, we only retrieve 6 items on the featured list, but if we ever change it in the future we would be covered.

Reply:
But we can unify the two maps in one step, what do you think?

```javascript
const displayedFeaturedSlugsMap = new Map(
  featuredPlugins
    .slice( 0, SHORT_LIST_LENGTH ) // only displayed plugins
    .map( ( plugin ) => [ plugin.slug, plugin.slug ] )
);

```

Reply:
>My main idea with creating a map was to build an O(n) instead of an O(n^2) solution

That's true! Yes, unifying them seems good.